common/utils.js

```diff
@@ -440,26 +440,35 @@ const getMultisigProof = async (config, chain, multisigSessionId) => {
 
 const calculateDomainSeparator = (chain, router, network) => keccak256(Buffer.from(`${chain}${router}${network}`));
 
-const getItsEdgeContract = (chainConfig) => {
-    const itsEdgeContract =
+const getItsChains = (chainConfig) => {
+    const isItsEdgeContract =
         chainConfig.contracts.InterchainTokenService?.address ||
-        chainConfig.contracts.ITS?.objects?.ChannelId ||
+        chainConfig.contracts.ITS?.objects?.ChannelId || // TODO remove this once devnet is redeployed
+        chainConfig.contracts.InterchainTokenService?.objects?.ChannelId ||
         chainConfig.contracts.interchain_token_service?.address;
 
-    if (!itsEdgeContract) {
+    return isItsEdgeContract;
+};
+
+const itsEdgeContract = (chainConfig) => {
+    const isItsEdgeContract = getItsChains(chainConfig);
+
+    if (!isItsEdgeContract) {
         throw new Error(`Missing InterchainTokenService edge contract for chain ${chainConfig.name}`);
     }
 
-    return itsEdgeContract;
+    return isItsEdgeContract;
 };
 
-const getItsEdgeChains = (config, excludeChainName) => {
-    return Object.keys(config.chains).filter((chain) => getItsEdgeContract(config.chains[chain]) && chain !== excludeChainName);
+const getItsEdgeChains = (config) => {
+    const itsChains = [];
+    Object.keys(config.chains).filter((chain) => getItsChains(config.chains[chain]) && itsChains.push(config.chains[chain].axelarId));
```

Comment:
weird mix of functional and imperative code
```suggestion
    Object.values(config.chains).filter((chain) => getItsEdgeContract(chain)).map((chain) => chain.axelarId);
```